weather.go

```diff
@@ -1,5 +1,138 @@
 package weather
 
-func Get(location, token string) ([]byte, error) {
-	return nil, nil
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type Weather struct {
+	Name    string
+	Weather []struct {
+		ID          int
+		Main        string
+		Description string
+		Icon        string
+	}
+	Main struct {
+		Temp      float64
+		FeelsLike float64
+		TempMin   float64
+		TempMax   float64
+		Humidity  int
+	}
+}
+
+type CityUnknown struct {
+	Cod     string
+	Message string
+}
+
+type Conditions struct {
+	City    string
+	OneWord string
+	Celsius float64
+}
+
+func CliOutput(token string, args []string) (output string) {
+
+	location := LocationFromArgs(args)
+	url := BuildURL(token, location)
+	callURL := CallURL(url)
+	conditions, err := Get(callURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	output = fmt.Sprintf("city: %s\nweather: %s\nCelsius: %v\n", conditions.City, conditions.OneWord, conditions.Celsius)
+
+	return output
+}
+
+func LocationFromArgs(args []string) string {
+
+	if len(os.Args) == 1 {
+		fmt.Fprintf(os.Stderr, "please set a location e.g. london\n")
+		os.Exit(2)
+	}
+
+	location := strings.Join(args[1:], "%20")
+
+	return location
+}
+
+func BuildURL(token string, location string) string {
+
+	domain := "api.openweathermap.org"
```

Comment:
This is going to make it difficult for us to inject the URL of a test HTTP server, isn't it? It would be convenient to have some kind of 'client object' we could construct which knows the URL of the weather API. Then in the test, you could construct this object, but set its `BaseURL` field (for example) to the test server's URL.